clients/cobol-lsp-vscode-extension/src/commands/RunAnalysisCLI.ts

```diff
@@ -0,0 +1,260 @@
+    const serverPath = this.getExtensionPath() + "/server/native";
+    const result = this.getServerPath(serverPath, platform);
+
+    if (result === "") {
+      return "";
+    }
+
+    return result + " " + this.buildAnalysisCommandPortion(currentFileLocation);
+  }
+
+  protected getServerPath(serverPath: string, platform: string) {
+    switch (platform) {
+      case "win32":
+        return serverPath;
+      case "linux":
+        return (serverPath += "/server-linux");
+      case "darwin": // macOS
+        return (serverPath += "/server-mac");
+      default: // Only Windows, Linux and Mac are supported.
+        return "";
+    }
+  }
+
+  /**
+   * Creates the command to run using the java build.
+   * @param currentFileLocation - Location of the main cobol file being analyzed.
+   * @protected
+   */
+  protected buildJavaCommand(currentFileLocation: string) {
+    const extensionFolder: string | undefined =
+      this.getExtensionPath() + "/server/jar/server.jar";
+
+    if (extensionFolder && currentFileLocation !== "") {
+      return (
+        'java -jar "' +
+        extensionFolder +
+        '" ' +
+        this.buildAnalysisCommandPortion(currentFileLocation)
+      );
+    }
+
+    return "";
+  }
+
+  /**
+   * Provides the portion of the command from "analysis" onwards.
+   * Is the same for both the Java and Native builds.
+   * @param currentFileLocation - Location of the main cobol file being analyzed.
+   * @protected
+   */
+  protected buildAnalysisCommandPortion(currentFileLocation: string) {
+    const copyBookCommand = `-cf=${
+      this.copybookConfigLocation === ""
+        ? "."
+        : '"' + this.copybookConfigLocation + '"'
+    }`;
+
+    return 'analysis -s "' + currentFileLocation + '" ' + copyBookCommand;
+  }
+
+  /**
+   * Sends a given command to a terminal.
+   * Checks to see if one named "Analysis" is already created, if so clear and reuse it.
+   * @param command - The command to run from the terminal.
+   * @protected
+   */
+  protected sendToTerminal(command: string) {
+    const existingTerminal = vscode.window.terminals.find(
+      (term: Terminal) => term.name === "Analysis",
+    );
+    const terminal = existingTerminal
+      ? existingTerminal
+      : vscode.window.createTerminal("Analysis");
+
+    if (existingTerminal) {
+      terminal.sendText(this.getClearCommand(process.platform));
+    }
+
+    terminal.sendText(command);
+    terminal.show(true);
+  }
+
+  /**
+   * Retrieves the location of the currently open file.
+   * If it is not saved, creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected getCurrentFileLocation() {
+    if (vscode.window.activeTextEditor) {
+      if (
+        vscode.window.activeTextEditor.document.uri.path
+          .toString()
+          .startsWith("Untitled")
```

Comment:
I am not sure if it is actually guaranteed that all the untitled documents will be following this naming convention.
```suggestion
        vscode.window.activeTextEditor.document.uri.scheme === "untitled"
```

Reply:
Implemented in latest commit.